bee-message/src/address/mod.rs

```diff
@@ -0,0 +1,88 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+mod ed25519;
+
+pub use ed25519::{Ed25519Address, ED25519_ADDRESS_LENGTH};
+
+use crate::{error::ValidationError, signature::SignatureUnlock};
+
+use bech32::{self, FromBase32, ToBase32, Variant};
+use bee_packable::{Packable, SliceUnpacker};
+
+use alloc::{str::FromStr, string::String};
+use core::convert::TryFrom;
+
+/// A generic address supporting different address kinds.
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Ord, PartialOrd, Hash, Packable)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Serialize, serde::Deserialize),
+    serde(tag = "type", content = "data")
+)]
+#[packable(tag_type = u8)]
+pub enum Address {
+    /// An Ed25519 address.
+    #[packable(tag = 0)]
+    Ed25519(Ed25519Address),
+}
+
+impl Address {
+    /// Returns the address kind of an `Address`.
+    pub fn kind(&self) -> u8 {
+        match self {
+            Self::Ed25519(_) => Ed25519Address::KIND,
+        }
+    }
+
+    /// Tries to create an `Address` from a Bech32 encoded string.
+    pub fn try_from_bech32(addr: &str) -> Result<Self, ValidationError> {
+        match bech32::decode(addr) {
+            Ok((_hrp, data, _)) => {
+                let bytes = Vec::<u8>::from_base32(&data).map_err(|_| ValidationError::InvalidAddress)?;
+                let mut unpacker = SliceUnpacker::new(bytes.as_slice());
```

Comment:
can't you use the magic method you introduced to `Packable` before? I forgot the name.

Reply:
Ha, looks like I missed this one!